postgres/src/app.py

```diff
@@ -0,0 +1,73 @@
+import logging
+
+from fastapi import Depends, FastAPI, HTTPException, status
+from sqlalchemy import text
+from sqlalchemy.exc import IntegrityError
+
+from src.db import get_db
+from src.model import Task, Tasks
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+app = FastAPI()
+
+
+@app.post("/tasks", status_code=status.HTTP_201_CREATED)
+async def create_task(task: str, db=Depends(get_db)) -> Task:
+    try:
+        query = text("INSERT INTO tasks (task) VALUES (:task) RETURNING id")
+        result = db.execute(query, {"task": task})
+        db.commit()
+        task_id = result.fetchone()[0]
+    except IntegrityError as e:
+        logger.warning(f"Error in creating task: {e.args[0]}")
+        if "psycopg2.errors.UniqueViolation" in e.args[0]:
+            raise HTTPException(
+                status_code=status.HTTP_400_BAD_REQUEST, detail="Task already exists"
+            )
+        else:
+            raise HTTPException(
```

Comment:
You can define a middleware to catch all the unanticipated errors. That way you won't need to clutter your code for every instance where some unexpected error may occur.

Reply:
For inspiration - checkout the template that you should have.